01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md

```diff
@@ -0,0 +1,607 @@
+
+- [Install and configure IntelliJ IDEA to access WSL](#install-and-configure-intellij-idea-to-access-wsl)
+
+### Install and configure IntelliJ IDEA to access WSL
+
+> [!NOTE]
+>
+> Follow these instructions when requested in the course. If you are following
+> this guide for the first time, you can skip this section.
+
+Inspired by the
+[How to use WSL development environment in JetBrains products by JetBrains](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html)
+guide.
+
+#### Install Java in WSL
+
+Install Java as mentioned in the course material. Install it in your Ubuntu
+distribution.
+
+> [!TIP]
+>
+> You might need to install the missing packages for
+> [SDKMAN!](https://sdkman.io/) to work correctly.
+>
+> Use `apt` to install them by running the following command in the terminal:
+>
+> ```bash
+> # Install the missing packages for SDKMAN!
+> sudo apt install <name of the missing package>
+> ```
+
+#### Install Maven in WSL
+
+Install Maven as mentioned in the course material. Install it in your Ubuntu
+distribution.
+
+#### Install IntelliJ IDEA
+
+Install IntelliJ IDEA as mentioned in the course material. Install it on your
+Windows.
+
+#### Configure Windows Firewall to allow IntelliJ IDEA to access WSL
+
+In order to use IntelliJ IDEA with WSL, you need to allow IntelliJ IDEA to
+access the WSL distribution through the Windows Firewall.
+
+> [!TIP]
+>
+> These instructions are meant to be used with IntelliJ IDEA. If you are using
+> another JetBrains IDE, you will need to adapt the instructions accordingly.
+
+Allow IntelliJ IDEA to access the WSL distribution through the Windows Firewall
+by running the following command in a PowerShell terminal as an administrator as
+seen in the previous section:
+
+```powershell
+# Allow IntelliJ IDEA to access the WSL distribution through the Windows Firewall
+New-NetFirewallRule -DisplayName "WSL" -Direction Inbound  -InterfaceAlias "vEthernet (WSL)"  -Action Allow
+
+# Renew the firewall rules
+Get-NetFirewallProfile -Name Public | Get-NetFirewallRule | where DisplayName -ILike "IntelliJ IDEA*" | Disable-NetFirewallRule
+```
+
+![Allow IntelliJ IDEA access to WSL](./images/windows-firewall.png)
+
+#### Create a new project in IntelliJ IDEA
+
+> [!CAUTION]
+>
+> If you have any issues with the following instructions, please check that you
+> have excluded WSL from Windows Defender. It is a known issue that Windows
+> Defender can block IntelliJ IDEA from accessing WSL:
+> [Impossible to configure SDK on windows with WSL](https://youtrack.jetbrains.com/issue/IDEA-273533/Impossible-to-configure-SDK-on-windows-with-WSL).
+
+Once you create a new project, IntelliJ IDEA should automatically detect the
+Java and Maven installations in WSL.
+
+Create a new project in your WSL home directory using the same instructions as
+in the course material.
+
+> [!CAUTION]
+>
+> Always create your projects in your WSL home directory to avoid file
+> permission issues!
```

Comment:
That's a very good point. 

Perhaps I'll add some further explanations or an extra chapter with the creation of a `~/devs/dai` folder, for example, and then the creation of a Java / Maven Hello world project in order to : 

1. Make sure everything works.
2. Force students to create their projects in a specific dev folder with a default structure for all future HEIG-VD courses. I think it's a good thing to be fairly strict on this point, in order to teach good practices right from the start and prevent students from storing their sources on OneDrive, Dropbox or anywhere else on the OS.

We might also add some examples of the basic commands (ls, cd etc...) to move inside the file tree and then compile and launch the hello world project with java, javac

Reply:
Very valid points. I have added some instructions to the guide and added a separated "Considerations for a development environment" guide to address some of these issues.